### PR TITLE
plugins.vimeo: add support for vimeo.com

### PR DESCRIPTION
**Pull request by melmorabity**
This plugin adds support for Vimeo. It supports both VOD and live streams. It also supports subtitles.

### CODE REVIEW

**Comment by gravyboat** [comment]
Since when does Vimeo do live streams? Can you link to some of them? We've been trying to take the stance of not allowing sites which are mostly VODs since we updated the plugin request requirements a while back so I'm hesitant to add Vimeo.

**Comment by back-to** [comment]
they got live streams

https://vimeo.com/blog/post/vimeo-live-is-here-professional-live-streaming

but they are rare because a Premium Plan is required

https://vimeo.com/upgrade?vcid=31608

File: docs/plugin_matrix.rst [hunk 1]
```diff
@@ -273,6 +273,7 @@ zattoo                  - zattoo.com         Yes   Yes
 zdf_mediathek           zdf.de               Yes   Yes   Streams may be geo-restricted to Germany.
 zengatv                 zengatv.com          Yes   No
 zhanqi                  zhanqi.tv            Yes   No
+vimeo                   vimeo.com            Yes   Yes   Password-protected videos are not supported.
```
Comment (back-to) [inline docs/plugin_matrix.rst]: alphabetical order

File: src/streamlink/plugins/vimeo.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        PluginArgument(
+            "mux-subtitles",
+            action="store_true",
+            help="Automatically mux available subtitles in to the output stream.",
+        )
+    )
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return Vimeo._url_re.match(url)
```
Comment (back-to) [inline src/streamlink/plugins/vimeo.py]:
```py
return cls._url_re.match(url)
```

File: docs/plugin_matrix.rst (hunk 1 above)
Comment (melmorabity) [reply docs/plugin_matrix.rst]: Good catch, fixed

File: src/streamlink/plugins/vimeo.py (hunk 2 above)
Comment (melmorabity) [reply src/streamlink/plugins/vimeo.py]: Fixed too

File: src/streamlink/plugins/vimeo.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        for stream_type in ("hls", "dash"):
+            if not stream_type in videos:
+                continue
+            for _, video_data in videos[stream_type]["cdns"].items():
+                url = video_data.get("url")
+                if stream_type == "hls":
+                    for stream in HLSStream.parse_variant_playlist(self.session, url).items():
+                        streams.append(stream)
+                elif stream_type == "dash":
+                    url = url.replace("master.json", "master.mpd")
```
Comment (back-to) [inline src/streamlink/plugins/vimeo.py]: this doesn't work with the livestream

```
error: Unable to parse XML: not well-formed (invalid token): line 1, column 0 (b'{"url":"https://live.vimeocdn.com ...)
```

url is `https://live-api.cloud.vimeo.com/sessions/.../playlist/dash.mpd?json=1&~id=player&~exp=&~sig=`

and the output is a json file with a dash url

`{"url":"https://live.vimeocdn.com/.../.../dash.mpd"}`

**Comment by gravyboat** [comment]
Thanks @melmorabity and @back-to!

**Pull request merged**